File: common/src/main/java/wile/redstonepen/platform/INetworkingPlatform.java
package wile.redstonepen.platform;

import net.minecraft.server.level.ServerLevel;
import net.minecraft.server.level.ServerPlayer;

public interface INetworkingPlatform
{
  void registerPayloads();
  void sendToPlayer(ServerPlayer player, wile.redstonepen.net.Networking.UnifiedPayload payload);
  void sendToAllPlayers(ServerLevel world, wile.redstonepen.net.Networking.UnifiedPayload payload);
}

File: common/src/main/kotlin/wile/redstonepen/platform/INetworkingPlatform.kt
package wile.redstonepen.platform

import net.minecraft.server.level.ServerLevel
import net.minecraft.server.level.ServerPlayer
import wile.redstonepen.net.Networking.UnifiedPayload

interface INetworkingPlatform {
    fun registerPayloads()

    fun sendToPlayer(player: ServerPlayer, payload: UnifiedPayload)

    fun sendToAllPlayers(world: ServerLevel, payload: UnifiedPayload)
}
